pkg/webhook/nodenetworkconfigurationpolicy/validation_test.go

```diff
@@ -0,0 +1,95 @@
+package nodenetworkconfigurationpolicy
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	shared "github.com/nmstate/kubernetes-nmstate/api/shared"
+	nmstatev1beta1 "github.com/nmstate/kubernetes-nmstate/api/v1beta1"
+)
+
+var _ = Describe("NNCP Conditions Validation Admission Webhook", func() {
+	nonProgressingPolicyConditions := shared.ConditionList{}
+	nonProgressingPolicyConditions.Set(
+		shared.NodeNetworkConfigurationPolicyConditionDegraded,
+		corev1.ConditionFalse,
+		shared.NodeNetworkConfigurationPolicyConditionSuccessfullyConfigured,
+		"",
+	)
+	nonProgressingPolicyConditions.Set(
+		shared.NodeNetworkConfigurationPolicyConditionAvailable,
+		corev1.ConditionTrue,
+		shared.NodeNetworkConfigurationPolicyConditionSuccessfullyConfigured,
+		"",
+	)
+	progressingPolicyConditions := shared.ConditionList{}
+	progressingPolicyConditions.Set(
+		shared.NodeNetworkConfigurationPolicyConditionDegraded,
+		corev1.ConditionUnknown,
+		shared.NodeNetworkConfigurationPolicyConditionConfigurationProgressing,
+		"",
+	)
+	progressingPolicyConditions.Set(
+		shared.NodeNetworkConfigurationPolicyConditionAvailable,
+		corev1.ConditionUnknown,
+		shared.NodeNetworkConfigurationPolicyConditionConfigurationProgressing,
+		"",
+	)
```

Comment:
Let's make policyconditions setters public same as we have at enactments so we can use them here, let's also hope it does not introduce cycles.

Reply:
Done